Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -166,6 +177,36 @@ namespace AzToolsFramework
             }
         }
 
+        void PrefabSystemComponent::SourceFileChanged(AZStd::string relativePath, AZStd::string scanFolder, [[maybe_unused]] AZ::Uuid sourceUUID)
+        {
+            auto found = m_templateFilePathToIdMap.find(relativePath.c_str());
+            if (found != m_templateFilePathToIdMap.end())
+            {
+                m_prefabLoader.ReloadTemplateFromFile(relativePath.c_str());
+                PropagateTemplateChanges(found->second);
+            }
+        }
+
+        void PrefabSystemComponent::SourceFileRemoved(
+            AZStd::string relativePath, AZStd::string scanFolder, [[maybe_unused]] AZ::Uuid sourceUUID)
+        {
+            auto found = m_templateFilePathToIdMap.find(relativePath.c_str());
+            if (found != m_templateFilePathToIdMap.end())
+            {
+                RemoveTemplate(found->second);
+
+                PrefabEditorEntityOwnershipInterface* prefabEditorEntityOwnershipInterface =
+                    AZ::Interface<PrefabEditorEntityOwnershipInterface>::Get();
+
+                if (prefabEditorEntityOwnershipInterface != nullptr)
+                {
+                     TemplateId rootTemplateId = prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
+                     PropagateTemplateChanges(rootTemplateId);
+                }
+                
+            }
+        }
```

Comment:
Should we really remove the template from memory if the file is deleted?
What I'd expect most DCC tools to do if the source file for something that is loaded is deleted is to still keep the file in memory, just mark it as dirty.

Reply:
Removing it from memory is the clean operation to do here. What use cases are you thinking of where keeping the template would be useful?

Reply:
What if the user deletes the prefab level file from disk? What state does that put the Editor in?

In general, though, I'd bring the topic up with UI (@rainbj) because I feel like the expected behavior from a UX perspective is that the prefab would stay. The fact that we're missing additional UI to show the information of which prefabs are loaded is a different issue :)

Reply:
Yes, levels require additional ui around deletions. Same applies to any modified prefab template that gets deleted. We should probably give an option to the users to save a backup or ignore the deletion or ignore their local changes. Currently these changes doesn't take any user input into consideration. They are barebone apis that need to be improved to include changes like conflict detection, user input, etc. Since these changes will be staged behind a development flag, I think they can go in in their current state and adjusted accordingly going forward.

Reply:
I understand the lack of UI, but I do believe the default behavior for file deletions should be to keep, not remove. 
It's ok for this to go in right now, I guess, just wanted to make that clear :D

Reply:
Just to tie a knot on this topic, if a user gives explicit permission to ignore local changes, then it should be fine to remove the template then right? I can't think of any reason for keeping it in-memory when user says its okay to be deleted.

Reply:
Correct, I just think the default behavior should be to keep. Since we don't have the UI yet, it would make sense to implement just the default behavior first. But since it's a WIP, it's ok as long as we're aligned on the end result.

To clarify further, I think the right behavior is to do what Notepad++ does in these situations:
![FileDeletion](https://user-images.githubusercontent.com/82231674/184702796-40291ab7-ed41-4cfc-aab9-e2350fde24bc.gif)

When an open file is deleted, the user is asked whether that file should be kept in memory or closed. The default option is to keep. What we could add is a "remember this" checkbox, but aside from that, I think this flow makes sense.

The only thing we don't really support at the moment in the Editor is closing a level - the only way you can have the Editor open with no level loaded is at startup, and we don't really support reverting to that state. I think it's technically doable though, and I think it would make sense if we wanted to move in that direction.

Reply:
I agree with @AMZN-daimini that it should be something like MarkAsDirty here eventually. However, without UI at this time, we can just really remove it to avoid unexpected issues that might come.

Reply:
There won't be any unexpected issues here since this feature will be behind a development flag. Plus, we need to define a default till the ui is finalized. I'd rather have this as a starting point and be confident that removal can work rather than have it do nothing, which is the current behavior. We can always fine-tune it as needed.

Reply:
Yes, I agree with the removal as the default behavior at this time.